### PR TITLE
chore(discoveryengine): update search sample to disable auto paginate

### PR DESCRIPTION
**Pull request by chenlei1216**
## Description

By default NodeJS has autoPaginate turned on, this will result in us iterate through all pages of search results, instead of just a single page restricted by pageSize.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [ ] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: discoveryengine/search.js [hunk 1]
```diff
@@ -57,10 +57,20 @@ async function main(
       servingConfig: name,
     };
 
+    const IResponseParams = {
+      ISearchResult: 0,
+      ISearchRequest: 1,
+      ISearchResponse: 2,
+    };
```
Comment (holtskinner) [inline discoveryengine/search.js:64]: Is there a built in enum in the client library for this?

If so, use it instead.

**Comment by holtskinner** [comment]
General Comment - I don't think it makes sense to change the default setting for this to not paginate, because there could be a large number of search results, causing the response to be large/time consuming. Maybe add a comment for this, or make it a parameter to the sample function to show how to change this.

**Review by holtskinner — CHANGES_REQUESTED** [review]

File: discoveryengine/search.js (hunk 1 above)
Comment (chenlei1216) [reply discoveryengine/search.js:64]: I can't find any. Will have to go with this first and update later if I can find better options.

**Comment by chenlei1216** [comment]
> General Comment - I don't think it makes sense to change the default setting for this to not paginate, because there could be a large number of search results, causing the response to be large/time consuming. Maybe add a comment for this, or make it a parameter to the sample function to show how to change this.

To clarify, we are still doing the pagination (pageSize is still being used). This is a special behavior for NodeJS that by default it's turning the API call result into an iterable of SearchResults by default (while for Python etc. it's only returning SearchResponse). This may cause several issues for our case, e.g. if users set page size as 10, and there are 100 results, NodeJS library by default will try to iterate through all 100 results by sending 10 search requests in sequence (offset=0, offset=10, offset=20...) until it exhausts the full list. If you have 1K, 10K search results this could take forever.

IMO we should always turn off the autoPaginate feature for the search use case, so we can get the single SearchResponse just for the page, in the above example it'll only return 10 documents. Hope this makes sense.

**Review by holtskinner — APPROVED** [review]

**Review by pattishin — APPROVED** [review]

**Pull request merged**